gcc/rust/checks/errors/rust-feature.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2020-2022 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef RUST_FEATURE_H
+#define RUST_FEATURE_H
+
+#include "rust-session-manager.h"
+#include "rust-optional.h"
+
+namespace Rust {
+
+struct Feature
+{
+  enum class State
+  {
+    ACCEPTED,
+    ACTIVE,
+    REMOVED,
+    STABILIZED,
+  };
+
+  Feature (State S, const char *N, const char *RS, const char *GS, uint64_t I,
+	   const Optional<CompileOptions::Edition> &E, const char *D)
+    : state (S), name (N), rustc_since (RS), gccrs_since (GS), issue (I),
+      edition (E), description (D)
```

Comment:
As mentioned on the other PR

1. In my opinion the parameters are not named appropriately and would benefit from a more descriptive name
2. I think we should remove the `gccrs_since` argument and field in the class